lib/inc/drogon/drogon_test.h

```diff
@@ -671,21 +671,21 @@ static int run(int argc, char** argv)
         on_leaving;                                         \
     } while (0);
 
-#define EVAL_AND_CHECK_TRUE__(func_name, expr)                       \
-    do                                                               \
-    {                                                                \
-        bool drresult__;                                             \
-        std::string drexpansion__;                                   \
-        std::tie(drresult__, drexpansion__) =                        \
-            (drogon::test::internal::Decomposer() <= expr).result(); \
-        if (!drresult__)                                             \
-        {                                                            \
-            ERROR_MSG(func_name, #expr)                              \
-                << "With expansion\n"                                \
-                << "  \033[0;33m" << drexpansion__ << "\x1B[0m\n\n"; \
-        }                                                            \
-        else                                                         \
-            SET_TEST_SUCCESS__;                                      \
+#define EVAL_AND_CHECK_TRUE__(func_name, expr)                         \
+    do                                                                 \
+    {                                                                  \
+        bool drresult__;                                               \
+        std::string drexpansion__;                                     \
+        std::tie(drresult__, drexpansion__) =                          \
+            (drogon::test::internal::Decomposer() <= (expr)).result(); \
```

Comment:
Decomposer depends on that C/C++ inserts the entire expression and evaluates from left to right to extract values within a expression. For example, `Decompose() <= 1 == 1` runs as `((Decompose() <= 1) == 1)`. Then it proxies the value for the real comparison. Putting () around `expr` breaks it.

Reply:
Should we suppress the warning for all CHECK blocks? Or other solutions? @an-tao @pkovacs

Reply:
I reverted that change, as requested.   How might we clear this class of warning (besides suppression):

```
/home/phil/Fedora/drogon/drogon-1.6.0/lib/inc/drogon/drogon_test.h:680:51: warning: suggest parentheses around comparison in operand of '==' [-Wparentheses]
  680 |           ((drogon::test::internal::Decomposer()) <= expr).result(); \
```

Reply:
I can't think of a solution off my head right now. It's unlikely braking an expression apart won't cause some sort of compiler warning. Since it's usually a sign of bugs. [Catch2's decomposer](https://github.com/catchorg/Catch2/blob/2c269eb6332bc1dd29047851ae1efc3cd4c260d2/src/catch2/internal/catch_test_macro_impl.hpp#L56) also uses the same construct.